lightningd/options.c

```diff
@@ -141,13 +153,14 @@ static char *opt_set_s32(const char *arg, s32 *u)
 static char *opt_add_ipaddr(const char *arg, struct lightningd *ld)
 {
 	size_t n = tal_count(ld->wireaddrs);
+	u16 defport = get_portnum(ld);
 	char const *err_msg;
 
 	assert(arg != NULL);
 
 	tal_resize(&ld->wireaddrs, n+1);
 
-	if (!parse_wireaddr(arg, &ld->wireaddrs[n], ld->portnum, &err_msg)) {
+	if (!parse_wireaddr(arg, &ld->wireaddrs[n], defport, &err_msg)) {
```

Comment:
It seems, if we do something like the below in our config file:

    ipaddr=127.0.0.1
    port=9999

The `wireaddr` will be `127.0.0.1:9735`?  Because `ipaddr` came before the `port`.  Perhaps we should make this clear in our documentation that `port` should be set before `ipaddr`, or explicitly provide the port in `ipaddr` setting.

Reply:
You're right: we need to use port there if it's set.

Reply:
After consideration, this whole option mess needs more work.  I've dropped this patch, and will revisit in another PR.